group-coordinator/src/main/java/org/apache/kafka/coordinator/group/Group.java

```diff
@@ -209,4 +219,50 @@ void validateOffsetFetch(
     default boolean shouldExpire() {
         return true;
     }
+
+    /**
+     * Computes the hash of the topics in a group.
+     *
+     * @param topicHashes The map of topic hashes. Key is topic name and value is the topic hash.
+     * @return The hash of the group.
+     */
+    static long computeGroupHash(Map<String, Long> topicHashes) {
```

Comment:
I wonder whether it is worth inlining the implementation from Guava or something similar to combine the hashes. It would avoid the extra collections. I am not sure whether it makes a real difference though. What are your thoughts?

Reply:
In the KIP, you also mentioned combining the index with the hash. Is this something done within `combineOrdered`?

Reply:
> I wonder whether it is worth inlining the implementation from Guava or something similar to combine the hashes. It would avoid the extra collections.
Yes, I copy some implementation to this function.

> In the KIP, you also mentioned combining the index with the hash. Is this something done within `combineOrdered`?
No, the `computeGroupHash` sorts topics by name and use this order to merge hashes. I also add test case `testComputeGroupHashWithDifferentOrder` and `testComputeGroupHashWithSameKeyButDifferentValue` to verify it.